core/state_processor_rip7560.go

```diff
@@ -15,6 +14,36 @@ import (
 	"strings"
 )
 
+const MAGIC_VALUE_SENDER = uint64(0xbf45c166)
+const MAGIC_VALUE_PAYMASTER = uint64(0xe0e6183a)
+const MAGIC_VALUE_SIGFAIL = uint64(0x31665494)
+const PAYMASTER_MAX_CONTEXT_SIZE = 65536
+
+func PackValidationData(authorizerMagic uint64, validUntil, validAfter uint64) []byte {
+
+	t := new(big.Int).SetUint64(uint64(validAfter))
+	t = t.Lsh(t, 48).Add(t, new(big.Int).SetUint64(validUntil&0xffffff))
+	t = t.Lsh(t, 160).Add(t, new(big.Int).SetUint64(uint64(authorizerMagic)))
+	return common.LeftPadBytes(t.Bytes(), 32)
+}
+
+func UnpackValidationData(validationData []byte) (authorizerMagic uint64, validUntil, validAfter uint64) {
+
+	t := new(big.Int).SetBytes(validationData)
+	authorizerMagic = t.Uint64()
+	validUntil = t.Rsh(t, 160).Uint64() & 0xffffff
+	validAfter = t.Rsh(t, 48).Uint64()
+	return
+}
+
+func UnpackPaymasterValidationReturn(paymasterValidationReturn []byte) (validationData, context []byte) {
+	validationData = paymasterValidationReturn[0:32]
+	//2nd bytes32 is ignored (its an offset value)
+	contextLen := new(big.Int).SetBytes(paymasterValidationReturn[64:96])
```

Comment:
Maybe this is a good place to check it for `MAX_CONTEXT_SIZE` ?
It will crash here if the returned array is not correct.

Reply:
actually, it isn't tested at all in this commit. added error checks in the paymaster PR